modules/yieldmoSyntheticInventoryModule.js

```diff
@@ -32,15 +56,274 @@ export function validateConfig(config) {
   }
 }
 
-function googletagCmd(config, containerName, googletag) {
-  const gamContainer = window.document.createElement('div');
+function googletagCmd(config, googletag) {
+  const gamContainer = window.top.document.createElement('div');
+  const containerName = 'ym_sim_container_' + config.placementId;
   gamContainer.id = containerName;
-  window.document.body.appendChild(gamContainer);
+  window.top.document.body.appendChild(gamContainer);
   googletag.defineSlot(config.adUnitPath, [1, 1], containerName)
     .addService(googletag.pubads())
     .setTargeting('ym_sim_p_id', config.placementId);
   googletag.enableServices();
   googletag.display(containerName);
 }
 
+function collectData(placementId, consentDataObj) {
+  const timeStamp = new Date().getTime();
+  const connection = window.navigator.connection || {};
+  const description = Array.prototype.slice.call(document.getElementsByTagName('meta'))
+    .filter((meta) => meta.getAttribute('name') === 'description')[0];
+  const pageDimensions = {
+    density: window.top.devicePixelRatio || 0,
+    height: window.top.screen.height || window.top.screen.availHeight || window.top.outerHeight || window.top.innerHeight || 481,
+    width: window.top.screen.width || window.top.screen.availWidth || window.top.outerWidth || window.top.innerWidth || 321,
+  };
+
+  return {
+    bust: timeStamp,
+    dnt: window.top.doNotTrack === '1' || window.top.navigator.doNotTrack === '1' || false,
+    pr: document.referrer || '',
+    _s: 1,
+    e: 4,
+    page_url: window.top.location.href,
+    p: placementId,
+    description: description ? description.content.substring(0, 1000) : '',
+    title: document.title,
+    scrd: pageDimensions.density,
+    h: pageDimensions.height,
+    w: pageDimensions.width,
+    pft: timeStamp,
+    ct: timeStamp,
+    connect: connection.effectiveType,
+    bwe: connection.downlink ? connection.downlink + 'Mb/sec' : '',
+    rtt: 'rtt' in connection ? String(connection.rtt) : undefined,
```

Comment:
what if `'rtt' in connection` but it's undefined? I think this can happen.
I suggest using 
`rtt: typeof connection.rtt !== "undefined" ? String(connection.rtt) : undefined,`
 or even 
`rtt: connection.rtt or connection.rtt === 0 ? String(connection.rtt) : undefined,`

Reply:
ok